src/components/ConnectToSageIntacctButton/index.tsx

```diff
@@ -0,0 +1,138 @@
+import React, {useRef, useState} from 'react';
+import type {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import Button from '@components/Button';
+import ConfirmModal from '@components/ConfirmModal';
+import * as Expensicons from '@components/Icon/Expensicons';
+import PopoverMenu from '@components/PopoverMenu';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import {removePolicyConnection} from '@libs/actions/connections';
+import Navigation from '@libs/Navigation/Navigation';
+import {getPoliciesConnectedToSageIntacct} from '@libs/PolicyUtils';
+import type {AnchorPosition} from '@styles/index';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type {PolicyConnectionName} from '@src/types/onyx/Policy';
+
+type ConnectToSageIntacctButtonProps = {
+    policyID: string;
+    shouldDisconnectIntegrationBeforeConnecting?: boolean;
+    integrationToDisconnect?: PolicyConnectionName;
+};
+
+function ConnectToSageIntacctButton({policyID, shouldDisconnectIntegrationBeforeConnecting, integrationToDisconnect}: ConnectToSageIntacctButtonProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+    const currentIntegration = translate('workspace.accounting.integrationName', integrationToDisconnect);
+    const integrationToConnect = translate('workspace.accounting.integrationName', CONST.POLICY.CONNECTIONS.NAME.SAGE_INTACCT);
+
+    const [isDisconnectModalOpen, setIsDisconnectModalOpen] = useState(false);
+
+    const [policies] = useOnyx(ONYXKEYS.COLLECTION.POLICY);
+    const hasPoliciesConnectedToSageIntacct = !!getPoliciesConnectedToSageIntacct(policies).length;
```

Comment:
Does it benefit if we move this to `actions/Policy` where we have `allPolicies` already from Onyx?

Reply:
This condition is now false when I connected then disconnected SageInacct.
Does this something we should fix from our BE?

Reply:
> This condition is now false when I connected then disconnected SageInacct.
Does this something we should fix from our BE?

I only use credentials from workspaces that are currently connected to Sage Intacct. When user disconnects from Sage intacct on some other policy, I don't save the credentials(they originally come from `policy.connections.intacct.config.credentials` object) . I also think they shouldn't be saved because it could lead to some situations like this:
<img width="414" alt="Screenshot 2024-06-27 at 10 38 24" src="https://github.com/Expensify/App/assets/88395093/ac18594d-83b8-481b-b3b8-b65e45d41301">

Reply:
AH got it!